diff --git a/node_modules/astro/dist/core/routing/rewrite.js b/node_modules/astro/dist/core/routing/rewrite.js
index a5bf382..d82e160 100644
--- a/node_modules/astro/dist/core/routing/rewrite.js
+++ b/node_modules/astro/dist/core/routing/rewrite.js
@@ -58,7 +58,7 @@ function findRouteToRewrite({
     if (route.pattern.test(decodedPathname)) {
       if (route.params && route.params.length !== 0 && route.distURL && route.distURL.length !== 0) {
         if (!route.distURL.find(
-          (url) => url.href.replace(outDir.toString(), "").replace(/(?:\/index\.html|\.html)$/, "") == trimSlashes(decodedPathname)
+          (url) => url.href.replace(outDir.toString(), "").replace(/(?:\/index\.html|\.html)$/, "") == trimSlashes(pathname)
         )) {
           continue;
         }
